share/p2p/peers/manager.go

```diff
@@ -148,14 +148,16 @@ func (m *Manager) Start(startCtx context.Context) error {
 	m.cancel = cancel
 
 	validatorFn := m.metrics.validationObserver(m.Validate)
-	err := m.shrexSub.AddValidator(validatorFn)
-	if err != nil {
-		return fmt.Errorf("registering validator: %w", err)
-	}
 
-	err = m.shrexSub.Start(startCtx)
-	if err != nil {
-		return fmt.Errorf("starting shrexsub: %w", err)
+	if !m.params.DisableShrexSub {
```

Comment:
minor style thought: this requires a little mental arithmetic, `if not false` double negative, which is more easily parsed as `true` and is a little oblique how it relies on . It might be nicer to read as `m.params.ShrexSubEnabled`, but i appreciate this would mean we'd have to be mindful of having a default of true for the parameter any time the params are set.

Reply:
Definitely agree and nit about this often, didn't catch it here so thanks 🙏